hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/StateNetworkInfo.java

```diff
@@ -113,8 +120,7 @@ public boolean containsNode(final long nodeId) {
 
     @Override
     public void updateFrom(@NonNull final State state) {
-        nodeInfos.clear();
-        nodeInfos.putAll(nodeInfosFrom(state));
+        nodeInfos = nodeInfosFrom(state);
```

Comment:
To play it safe, I think `nodeInfosFrom()` should return an unmodifiable `Map`. Wrapping the generated `Map` with `Collections.unmodifiableMap()` should be sufficient here.

Reply:
Done.